packages/venia-ui/lib/components/CheckoutPage/checkoutPage.js

```diff
@@ -0,0 +1,190 @@
+    const { classes: propClasses } = props;
+    const [
+        {
+            isGuestCheckout,
+            isCartEmpty,
+            shippingInformationDone,
+            shippingMethodDone
+        },
+        { handleSignIn, setShippingInformationDone, setShippingMethodDone }
+    ] = useCheckoutPage();
+
+    const classes = mergeClasses(defaultClasses, propClasses);
+
+    const renderIfGuestCheckout = useCallback(renderIf(isGuestCheckout), [
+        isGuestCheckout
+    ]);
+    const renderIfCartNotEmpty = useCallback(renderIf(!isCartEmpty), [
+        isCartEmpty
+    ]);
+    const renderIfShippingInformationDone = useCallback(
+        renderIf(shippingInformationDone)
+    );
+    const renderIfShippingMethodDone = useCallback(
+        renderIf(shippingMethodDone)
+    );
+
+    return (
+        <div className={classes.root}>
+            <Title>{`Checkout - ${STORE_NAME}`}</Title>
+            {/**
+             * Lazy rendering checkout page components only
+             * if the cart is not empty.
+             */
+            renderIfCartNotEmpty(
+                () => (
+                    <Fragment>
+                        {/**
+                         * Lazy rendering Guest checkout options
+                         * only if use is not signed in.
+                         */
+                        renderIfGuestCheckout(() => (
+                            <GuestCheckoutOptions
+                                classes={classes}
+                                handleSignIn={handleSignIn}
+                            />
+                        ))}
+                        <GreetingMessage
+                            isGuestCheckout={isGuestCheckout}
+                            classes={classes}
+                        />
+                        <div className={classes.body}>
+                            <div
+                                className={`${
+                                    classes.shipping_information_container
+                                } ${classes.section_container}`}
+                            >
+                                <ShippingInformation
+                                    onSave={setShippingInformationDone}
+                                    doneEditing={shippingInformationDone}
+                                />
+                            </div>
+                            {/**
+                             * Lazy rendering Shipping method component
+                             * only if shipping information component is
+                             * done editing.
+                             */
+                            renderIfShippingInformationDone(() => (
+                                <Fragment>
+                                    <div
+                                        className={`${
+                                            classes.shipping_method_container
+                                        } ${classes.section_container}`}
```

Comment:
Avoid concatenating classes like this. In `checkoutPage.css` you should compose the shipping method container class with the `section_container`

```css
.shipping_method_container {
  composes: section_container;
  //...other css specific to shipping method container
}
```

Reply:
Din't know about this approach. Nice. Will do that.